banananation/src/main/java/gov/nasa/jpl/aerie/banananation/Configuration.java

```diff
@@ -2,13 +2,15 @@
 
 import java.nio.file.Path;
 
+import gov.nasa.jpl.aerie.merlin.framework.annotations.ActivityType;
+
 public final record Configuration (int initialPlantCount, String initialProducer, Path initialDataPath) {
 
   public static final int DEFAULT_PLANT_COUNT = 200;
   public static final String DEFAULT_PRODUCER = "Chiquita";
   public static final Path DEFAULT_DATA_PATH = Path.of("/etc/os-release");
 
-  public static Configuration defaultConfiguration() {
+  public static @ActivityType.Template Configuration defaultConfiguration() {
```

Comment:
I'm not too fond of using an `ActivityType` annotation here so I'm open to naming suggestions. I could make a separate class of annotations under a `Configuration` annotation interface. However, since these are the same annotations that are seen in activities I think a better solution would be to have a single source for these, under a different annotation interface.

The trouble is I'm not sure what name to use to unify activity and configuration types. Within the annotation processing code I used "specification type" – maybe this would be okay?

Reply:
Activities and configurations both are "parameterized" – not in a programming language generic type of way – but in the sense that they both contain parameter members. I would use the term "parameterized type" here but that's an overloaded term which would cause confusion.

Reply:
I agree the `ActivityType` is misleading since this is a part of the configuration that is being overridden. I like the idea of doing something like `@Configuration.template`

Reply:
Definitely. My biggest issue with doing `@Configuration.Template` is that it opens to door to mission modelers accidentally using `@Configuration.Template` within an activity instead of `@ActivityType.Template`. We could of course issue a warning/error in this case but that's another check we'd have to write into the annotation processing code.

And since these annotations are for declaring the same behaviour it makes sense for them to have the same qualified name `@SomeGeneralTerm.Template` for example :)

Reply:
I'm not against duplicating the hierarchy for model configuration; it seems like the least smelly solution in the short term. Longer-term, I think there are some larger-scale refactorings that could be done to simplify this system and bring the various facets into better alignment.

Reply:
> it makes sense for them to have the same qualified name

I agree with your rationale as well, @pcrosemurgy. If it's not difficult, I definitely support unifying the annotation hierarchy for generating mappers.

Brainstorming some vocabulary words that apply to both activity mappers and model configuration mappers:

* Mapper (obviously)
* Binding (exposing/exporting model entities through the mission-agnostic interface)
* Glue
* Export
* Expose

"Export" is growing on me, honestly. Examples: `@Export.Template`; `@Export.Parameter`; `@Export.WithDefaults`; it acts like a command prefix, which feels quite nice in this context.

Reply:
I haven't looked closely at the PR to see if you've done this, but the `ActivityType.WithMapper` annotation isn't really a candidate for this kind of unification, I think. It probably works best as another top-level annotation under the `MissionModel` hierarchy.

Reply:
^ 100%, I only want activity types and configurations to have a unified annotation interface with `@Template`, `@WithDefaults`, and `@Parameter` annotations.

Reply:
>"Export" is growing on me

That's pretty good! We're "exporting" parameters and their defaults to planners.

Reply:
I think `@Validation` is also something that could be shared in common, although I don't believe it's functionality currently supported for mission configuration, so it's something that could be done at a later date. (One of those refactorings I mentioned.)

Reply:
For sure, validation as well! We have a ticket to support validating config. parameters just like activities.

Reply:
`@Export.*` change has been made as of the latest pushes :)